go/vt/vtgate/executor.go

```diff
@@ -645,6 +650,56 @@ func (e *Executor) handleSet(ctx context.Context, safeSession *SafeSession, sql
 	return &sqltypes.Result{}, nil
 }
 
+func (e *Executor) handleSetVitessMetadata(ctx context.Context, session *SafeSession, k sqlparser.SetKey, v interface{}) (*sqltypes.Result, error) {
+	val, ok := v.(string)
+	if !ok {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "unexpected value type for charset: %T", v)
+	}
+
+	ts, err := e.serv.GetTopoServer()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := ts.UpsertMetadata(ctx, k.Key, val); err != nil {
+		return nil, err
+	}
+
+	return &sqltypes.Result{}, nil
+}
+
+func (e *Executor) handleShowVitessMetadata(ctx context.Context, session *SafeSession, ksName string, opt *sqlparser.ShowTablesOpt) (*sqltypes.Result, error) {
+	ts, err := e.serv.GetTopoServer()
+	if err != nil {
+		return nil, err
+	}
+
+	var metadata map[string]string
+	if opt.Filter == nil {
+		metadata, err = ts.GetMetadata(ctx, "")
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		metadata, err = ts.GetMetadata(ctx, opt.Filter.Like)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	rows := make([][]sqltypes.Value, 0, 2)
```

Comment:
I'd use `len(metadata)` as the capacity here.

Reply:
Done